### PR TITLE
Fix Woff2 Glyph Loading

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/Fonts/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

Our Woff2 loader is not loading glyphs properly. See https://github.com/SixLabors/Fonts/pull/198#issuecomment-927122736

- [x] Fix loading of woff2 headers.
- [x] Fix loading of composite gyphs to return the same result as TTF version

### CODE REVIEW

**Comment by JimBobSquarePants** [comment]
Failing test is NET472 Vector2 accuracy. It's 224AM here so will leave for tomorrow to fix. (_unless someone wants to._)

**Comment by brianpopow** [comment]
@JimBobSquarePants great that you have found the issue with the composite glyphs!

> Failing test is NET472 `Vector2` accuracy. It's 2:24AM here so will leave for tomorrow to fix. (_unless someone wants to._)

Using tolerant comparer now: 648fc94a

File: src/SixLabors.Fonts/Tables/Woff2Utils.cs [hunk 1]
```diff
@@ -47,77 +47,102 @@ internal class Woff2Utils
...
+            byte flagsByte = reader.ReadByte();
+            int knownTable = flagsByte & 0x3F;
+            string tableName = knownTable == 0x3F ? reader.ReadTag() : KnownTableTags[knownTable];
 
-            if (!ReadUIntBase128(reader, out uint tableOrigLength))
+            uint flags = 0;
+            byte xformVersion = (byte)((flagsByte >> 6) & 0x03);
+
+            // 0 means xform for glyph/loca, non-0 for others
+            if (tableName is "glyf" or "loca")
```
Comment (brianpopow) [inline src/SixLabors.Fonts/Tables/Woff2Utils.cs:60]: minor: use const for the table names?

**Review by brianpopow — APPROVED** [review]
looks good

File: src/SixLabors.Fonts/Tables/Woff2Utils.cs (hunk 1 above)
Comment (JimBobSquarePants) [reply src/SixLabors.Fonts/Tables/Woff2Utils.cs:60]: I _think_ the compiler will just make these constants anyway.

**Pull request merged**